apiserver/facades/agent/provisioner/provisioninginfo.go

```diff
@@ -108,7 +108,14 @@ func (p *ProvisionerAPI) getProvisioningInfo(m *state.Machine, env environs.Envi
 		EndpointBindings:  endpointBindings,
 		ImageMetadata:     imageMetadata,
 		ControllerConfig:  controllerCfg,
-	}, nil
+	}
+
+	cloudInitUserData := env.Config().CloudInitUserData()
```

Comment:
Why not just do this above the ProvisioningInfo struct creating along with the other attributes. The return can then stay the same. 
env.Config().CloudInitUserData() should return a nil map if there's nothing defined. It might return an empty map but even then that's sort of ok. But nil better since the other things are nil if not set, eg machineEndpointBindings(m) returns nil. Actually, I think it does need to be nil so that the omitempty yaml tag works.

Reply:
updated